docs/2.8.0/docs/daml-finance/reference/patterns.rst

```diff
@@ -51,6 +51,35 @@ could in principle stay untouched.
 For an example where the Factory pattern is used, check out the
 :doc:`Holdings tutorial <../tutorials/getting-started/holdings>`.
 
+.. _reference-pattern:
+
+Reference pattern
+-----------------
+
+The Reference pattern is used to leverage the functionalities of :ref:`Contract Keys <contractkeys>`
+when working with interfaces.
+
```

Comment:
Maybe also clarify "There is currently no built-in support for interface keys, so this is used as a workaround", in order to specify why we are using this pattern and what would happen if we would not use it.